terraform/modules/matrix-service/infra/converter_batch.tf

```diff
@@ -0,0 +1,188 @@
+  description = "Role to Allow EC2 Spot Fleet to request and terminate Spot Instances on your behalf."
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "spotfleet.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy_attachment" "AmazonEC2SpotFleetRole" {
+  role = "${aws_iam_role.AmazonEC2SpotFleetRole.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2SpotFleetRole"
+}
+
+resource aws_iam_service_linked_role "AWSServiceRoleForEC2Spot" {
+  aws_service_name = "spot.amazonaws.com"
+  description = "Allows EC2 Spot to launch and manage spot instances."
+}
+
+resource aws_iam_service_linked_role "AWSServiceRoleForEC2SpotFleet" {
+  aws_service_name = "spotfleet.amazonaws.com"
+  description = "Default EC2 Spot Fleet Service Linked Role"
+}
+
+data "external" "converter_desired_vcpus" {
+  program = ["python", "${path.module}/fetch_batch_vcpus.py"]
+
+  query = {
+    compute_environment_name = "dcp-matrix-converter-cluster-${var.deployment_stage}"
+  }
+}
+
+resource "aws_batch_compute_environment" "converter_compute_env" {
+  compute_environment_name = "dcp-matrix-converter-cluster-${var.deployment_stage}"
+  type = "MANAGED"
+  service_role = "${aws_iam_role.AWSBatchServiceRole.arn}"
+  compute_resources {
+    type = "SPOT"
+    bid_percentage = 100
+    spot_iam_fleet_role = "${aws_iam_role.AmazonEC2SpotFleetRole.arn}"
+    max_vcpus = 256
+    min_vcpus = 0
+    // You must set desired_vcpus otherwise you get error: "desiredvCpus should be between minvCpus and maxvCpus"
+    // However this is actually not settable in AWS.  It will not let you change it.
+    // Here we use an external data source to dynamically set the desired vcpus to match current state.
+    desired_vcpus = "${data.external.converter_desired_vcpus.result.desired_vcpus}"
+    instance_type = [
+      "m4"
+    ]
+    image_id = "${var.converter_cluster_ami_id}"
+    subnets = ["${data.aws_subnet.default.*.cidr_block}"]
+    security_group_ids = [
+      "${aws_default_vpc.default.default_security_group_id}"
+    ]
+    ec2_key_pair = "matrix-service-${var.deployment_stage}"
+    instance_role = "${aws_iam_instance_profile.ecsInstanceRole.arn}"
+  }
+  depends_on = [
+    "aws_iam_role_policy_attachment.AWSBatchServiceRole"
+  ]
+}
+
+resource "aws_batch_job_queue" "converter_job_queue" {
+  name = "dcp-matrix-converter-queue-${var.deployment_stage}"
+  compute_environments = [
+    "${aws_batch_compute_environment.converter_compute_env.arn}"]
+  priority = 1
+  state = "ENABLED"
+}
+
+resource "aws_iam_policy" "converter_job_policy" {
+  name = "dcp-matrix-converter-job-${var.deployment_stage}"
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Sid": "s3",
+        "Effect": "Allow",
+        "Action": [
+          "s3:*"
```

Comment:
Do we want to restrict this?

Reply:
I think it should just need `s3:PutObject` if `s3:ListBucket` and `s3:GetObject` are open to everyone.

Reply:
sounds good